From 217613e9067dcfbd58ac3acc66f611026efd378f Mon Sep 17 00:00:00 2001
From: "yinbo.zhu" <yinbo.zhu@nxp.com>
Date: Wed, 26 Apr 2017 12:31:22 +0800
Subject: [PATCH 728/741] usb: udc: for fix 1021atwr compile issue

commit ec498dfbeed3ac05a8bae909e7bb344a2bba7a5c from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

replace dev_archdata.dma_ops to dev.dma_ops and remove
the powerpc configures for fix ls1021atwr compiled issue.

Signed-off-by: yinbo.zhu <yinbo.zhu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/usb/gadget/udc/fsl_udc_core.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/usb/gadget/udc/fsl_udc_core.c b/drivers/usb/gadget/udc/fsl_udc_core.c
index 6827874f6ae4..7925172eb49e 100644
--- a/drivers/usb/gadget/udc/fsl_udc_core.c
+++ b/drivers/usb/gadget/udc/fsl_udc_core.c
@@ -336,12 +336,6 @@ static int dr_controller_setup(struct fsl_udc *udc)
 	}
 #endif
 
-#if (defined(CONFIG_PPC32) || defined(CONFIG_PPC64)) && \
-!defined(CONFIG_NOT_COHERENT_CACHE)
-/* Turn on cache snooping hardware, since some PowerPC platforms
- * wholly rely on hardware to deal with cache coherent.
- */
-
 #if !defined(CONFIG_NOT_COHERENT_CACHE)
 	/* Turn on cache snooping hardware, since some PowerPC platforms
 	 * wholly rely on hardware to deal with cache coherent. */
@@ -2495,7 +2489,7 @@ static int fsl_udc_probe(struct platform_device *pdev)
 	/* Setup gadget.dev and register with kernel */
 	dev_set_name(&udc_controller->gadget.dev, "gadget");
 	udc_controller->gadget.dev.of_node = pdev->dev.of_node;
-	set_dma_ops(&udc_controller->gadget.dev, pdev->dev.archdata.dma_ops);
+	set_dma_ops(&udc_controller->gadget.dev, pdev->dev.dma_ops);
 
 	if (!IS_ERR_OR_NULL(udc_controller->transceiver))
 		udc_controller->gadget.is_otg = 1;
-- 
2.17.1

